test/harvest/hostmetrics_test.go

```diff
@@ -63,6 +63,77 @@ func TestHostCPU(t *testing.T) {
 		t.Logf("CPUPercents: %f, %f", beforeSample.CPUPercent, afterSample.CPUPercent)
 	})
 }
+func TestHostSharedMemory(t *testing.T) {
+	ctx := new(mocks.AgentContext)
+	ctx.On("Config").Return(&config.Config{
+		MetricsNetworkSampleRate: 1,
+	})
+	storageSampler := storage.NewSampler(ctx)
+
+	systemSampler := metrics.NewSystemSampler(ctx, storageSampler)
+
+	sampleB, _ := systemSampler.Sample()
+	beforeSample := sampleB[0].(*metrics.SystemSample)
+
+	f, err := os.Create("/dev/shm/test")
+	require.NoError(t, err)
+
+	for i := 0; i < 1024*1024; i++ {
+		f.Write([]byte("0"))
+	}
+
+	f.Sync()
+	defer func() {
+		require.NoError(t, f.Close())
+		os.Remove(f.Name())
+	}()
+
+	testhelpers.Eventually(t, timeout, func(st require.TestingT) {
+		sampleB, _ = systemSampler.Sample()
+		afterSample := sampleB[0].(*metrics.SystemSample)
+
+		assert.True(st, beforeSample.MemorySharedBytes+(1024*1024) == afterSample.MemorySharedBytes, "Shared Memory used did not increase enough, SharedMemoryBefore: %f SharedMemoryAfter %f ", beforeSample.MemorySharedBytes, afterSample.MemorySharedBytes)
```

Comment:
Out of curiosity, does it **always** eval to equal? wouldn't it be safer to do as in the other `<=` to avoid flaky tests?